### PR TITLE
Add Kay65 PCB

### PR DESCRIPTION
**Pull request by mechlovin**
## Description

Add Kay65 PCB. Top right RGB LED, Atmega32u4.
Thank you.
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mechlovin/kay65/rev1/rules.mk [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```
Comment (drashna) [inline keyboards/mechlovin/kay65/rev1/rules.mk:15]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/mechlovin/olly/jf/rules.mk [hunk 2]
```diff
@@ -10,7 +10,7 @@ BOOTLOADER = USBasp
 # Build Options
 #   change yes to no to disable
 #
-BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+BOOTMAGIC_ENABLE = lite      # Enable Bootmagic Lite
```
Comment (fauxpark) [inline keyboards/mechlovin/olly/jf/rules.mk:13]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/mechlovin/kay65/rev1/rules.mk [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+NKRO_ENABLE = no            # USB Nkey Rollover
```
Comment (fauxpark) [inline keyboards/mechlovin/kay65/rev1/rules.mk:16]:
```suggestion
NKRO_ENABLE = no            # Enable N-Key Rollover
```

File: keyboards/mechlovin/kay65/rev1/info.json [hunk 4]
```diff
@@ -0,0 +1,225 @@
+{
+    "keyboard_name": "Kay65",
+    "url": "",
+    "maintainer": "Mechlovin' Studio",
+    "width": 16,
+    "height": 5,
```
Comment (fauxpark) [inline keyboards/mechlovin/kay65/rev1/info.json:6]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/mechlovin/kay65/rev1/rev1.c [hunk 5]
```diff
@@ -0,0 +1,11 @@
+// Copyright %(YEAR)s %(YOUR_NAME)s (@%(USER_NAME)s)
```
Comment (fauxpark) [inline keyboards/mechlovin/kay65/rev1/rev1.c:1]: These should be filled out.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/mechlovin/kay65/rev1/rev1.h [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    { K10,   K11,   K12,   K13,   K14,   K15,   K16,   K17,   K18,   K19,   K1A,   K1B,   K1C,   KC_NO, K1E   }, \
+    { K20,   K21,   K22,   K23,   K24,   K25,   K26,   K27,   K28,   K29,   K2A,   K2B,   K2C,   K2D,   K2E   }, \
+    { K30,   KC_NO, K32,   K33,   K34,   K35,   K36,   K37,   K38,   K39,   K3A,   K3B,   K3C,   K3D,   K3E   }, \
+    { K40,   K41,   K42,   KC_NO, KC_NO, KC_NO, K46,   KC_NO, KC_NO, KC_NO, K4A,   K4B,   K4C,   K4D,   K4E   }, \
+}
+
+#define LAYOUT_65_iso_blocker( \
+    K00, K01, K02, K03, K04, K05, K06, K07, K08, K09, K0A, K0B, K0C, K0D, K0E,   \
+    K10, K11, K12, K13, K14, K15, K16, K17, K18, K19, K1A, K1B, K1C, K2D, K1E,   \
+    K20, K21, K22, K23, K24, K25, K26, K27, K28, K29, K2A, K2B, K2C,      K2E,   \
```
Comment (noroadsleft) [inline keyboards/mechlovin/kay65/rev1/rev1.h:53]:
```suggestion
    K10, K11, K12, K13, K14, K15, K16, K17, K18, K19, K1A, K1B, K1C,      K1E,   \
    K20, K21, K22, K23, K24, K25, K26, K27, K28, K29, K2A, K2B, K2C, K2D, K2E,   \
```

File: keyboards/mechlovin/kay65/rev1/info.json [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+                {"label":"K14 (B1,C1)", "x":4.5, "y":1},
+                {"label":"K15 (B1,C2)", "x":5.5, "y":1},
+                {"label":"K16 (B1,C3)", "x":6.5, "y":1},
+                {"label":"K17 (B1,C4)", "x":7.5, "y":1},
+                {"label":"K18 (B1,C5)", "x":8.5, "y":1},
+                {"label":"K19 (B1,C6)", "x":9.5, "y":1},
+                {"label":"K1A (B1,C7)", "x":10.5, "y":1},
+                {"label":"K1B (B1,D0)", "x":11.5, "y":1},
+                {"label":"K1C (B1,D1)", "x":12.5, "y":1},
+                {"label":"K2D (B2,D2)", "x":13.75, "y":1, "w":1.25, "h":2},
```
Comment (noroadsleft) [inline keyboards/mechlovin/kay65/rev1/info.json:180]:
```suggestion
```

File: keyboards/mechlovin/kay65/rev1/info.json [hunk 8]
```diff
@@ -0,0 +1,223 @@
...
+                {"label":"K24 (B2,C1)", "x":4.75, "y":2},
+                {"label":"K25 (B2,C2)", "x":5.75, "y":2},
+                {"label":"K26 (B2,C3)", "x":6.75, "y":2},
+                {"label":"K27 (B2,C4)", "x":7.75, "y":2},
+                {"label":"K28 (B2,C5)", "x":8.75, "y":2},
+                {"label":"K29 (B2,C6)", "x":9.75, "y":2},
+                {"label":"K2A (B2,C7)", "x":10.75, "y":2},
+                {"label":"K2B (B2,D0)", "x":11.75, "y":2},
+                {"label":"K2C (B2,D1)", "x":12.75, "y":2},
+                {"label":"K2E (B2,D3)", "x":15, "y":2},
```
Comment (noroadsleft) [inline keyboards/mechlovin/kay65/rev1/info.json:195]:
```suggestion
                {"label":"K2D (B2,D2)", "x":13.75, "y":1, "w":1.25, "h":2},
                {"label":"K2E (B2,D3)", "x":15, "y":2},
```

File: keyboards/mechlovin/kay65/rev1/rules.mk [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = no            # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
```
Comment (noroadsleft) [inline keyboards/mechlovin/kay65/rev1/rules.mk:18]:
```suggestion
AUDIO_ENABLE = no           # Audio output

LAYOUTS = 65_ansi_blocker 65_iso_blocker
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
Are the changes to the Olly JF intended to be in this PR?

File: keyboards/mechlovin/olly/jf/config.h [hunk 10]
```diff
@@ -72,3 +72,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 /* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
 #define DEBOUNCE 5
+
+/* Bootmagic Lite key configuration */
+#define BOOTMAGIC_LITE_ROW 1
+#define BOOTMAGIC_LITE_COLUMN 14
```
Comment (noroadsleft) [inline keyboards/mechlovin/olly/jf/config.h:78]: Based on this, [keyboards/mechlovin/olly/jf/readme.md#L23](https://github.com/qmk/qmk_firmware/blame/2b4eee4bf75f828360c4d874b3b6e0b4afa10101/keyboards/mechlovin/olly/jf/readme.md#L23) should read:

```
By bootloader: hold Backspace key while plugging in
```

File: keyboards/mechlovin/olly/jf/keymaps/via/config.h [hunk 11]
```diff
@@ -17,6 +17,5 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 #pragma once
 
-#define DYNAMIC_KEYMAP_EEPROM_MAX_ADDR 1607
-#define DYNAMIC_KEYMAP_LAYER_COUNT 6
+#define DYNAMIC_KEYMAP_LAYER_COUNT 2
```
Comment (noroadsleft) [inline keyboards/mechlovin/olly/jf/keymaps/via/config.h:20]: [keyboards/mechlovin/olly/jf/keymaps/via/keymap.c](https://github.com/qmk/qmk_firmware/blob/2b4eee4bf75f828360c4d874b3b6e0b4afa10101/keyboards/mechlovin/olly/jf/keymaps/via/keymap.c) still has six layers.

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by noroadsleft — CHANGES_REQUESTED** [review]
I maintain that due to the Bootmagic settings, [Line 23 of `keyboards/mechlovin/olly/jf/readme.md`](https://github.com/qmk/qmk_firmware/blame/fbec68f89898c264d995209bae068129517ab6df/keyboards/mechlovin/olly/jf/readme.md#L23) should read:

```
By bootloader: hold Insert key while plugging in
```

Everything else looks good though.

**Comment by mechlovin** [comment]
> I maintain that due to the Bootmagic settings, [Line 23 of `keyboards/mechlovin/olly/jf/readme.md`](https://github.com/qmk/qmk_firmware/blame/fbec68f89898c264d995209bae068129517ab6df/keyboards/mechlovin/olly/jf/readme.md#L23) should read:
> 
> ```
> By bootloader: hold Insert key while plugging in
> ```
> 
> Everything else looks good though.

Our apologies for not noticing your question regarding the Olly JF from 10 Jan. The changes to the Olly JF are isolated to that PCB only. Rationale for the changes is as follows:

- Currently, holding ESC + Plug cable enters Bootloader. We understand that you'd like to change it to Insert + Plug cable, however that cannot be changed without the use of a hardware programmer, which is not ideal as the Olly JF PCBs are already being used by many users. 

- We made a mistake by having both Bootloader and Bootmagic on ESC, and while Bootloader cannot be changed easily, changing the way to enter Bootmagic can be done in QMK. So that's why we're suggesting keeping ESC + Cable = Bootloader (as current setting); and making a change in QMK so that Insert + Cable = Bootmagic. 

We hope the explanation is clear enough and that the change we proposed is ok. 

With regards to the Kay65, everything is standard so if everything looks good, please go ahead with approval without worries. Thank you very much!

**Comment by noroadsleft** [comment]
@mechlovin Do I understand this correctly, then?

- `Esc` + plug in USB = jump to bootloader (cannot be changed without flashing a new bootloader using a hardware programmer)
- `Insert` + plug in USB = jump to bootloader *and* reset EEPROM via QMK's Bootmagic

**Comment by mechlovin** [comment]
>

**Comment by mechlovin** [comment]
> @mechlovin Do I understand this correctly, then?
> 
> * `Esc` + plug in USB = jump to bootloader (cannot be changed without flashing a new bootloader using a hardware programmer)
> * `Insert` + plug in USB = jump to bootloader _and_ reset EEPROM via QMK's Bootmagic

Yes that's correct! We hope that doesn't cause you any trouble.

**Comment by noroadsleft** [comment]
> Yes that's correct! We hope that doesn't cause you any trouble.

Ah, okay. Then an additional line in the readme to specify Bootloader Reset vs. Bootmagic Reset would be good to have.

File: keyboards/mechlovin/olly/jf/readme.md [hunk 12]
```diff
@@ -18,6 +18,7 @@ Flashing example for this keyboard:
 
 See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
 
-**Reset Key:** 2 ways to put the Olly JF into bootloader:
+**Reset Key:** 3 ways to put the Olly JF into bootloader:
 By keycode: Tap RESET keycode.
-By bootloader: hold ESC key while plugging in
\ No newline at end of file
+By Bootloader: hold ESC key while plugging in (jump to bootloader)
+By Bootmagic: hold Insert key while pluging in (jump to bootloader and resset EEPROM)
```
Comment (noroadsleft) [inline keyboards/mechlovin/olly/jf/readme.md:24]: Tiny typo here. :)

```suggestion
By Bootmagic: hold Insert key while pluging in (jump to bootloader and reset EEPROM)
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: keyboards/mechlovin/kay65/rev1/config.h [hunk 13]
```diff
@@ -0,0 +1,151 @@
...
+
+/* disable action features */
+//#define NO_ACTION_LAYER
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
+
+/* disable these deprecated features by default */
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
+
```
Comment (fauxpark) [inline keyboards/mechlovin/kay65/rev1/config.h:148]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by noroadsleft — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**